crd-generator/api/src/test/java/io/fabric8/crd/example/map/ContainingMapsSpec.java

```diff
@@ -32,4 +33,8 @@ public Map<String, Map<String, List<Boolean>>> getTest2() {
     return test2;
   }
 
+  private MyMultiMap<String, String> test3;
+
+  class MyMultiMap<K, V> extends HashMap<K, List<V>> {
```

Comment:
[The new implementation](https://github.com/sundrio/sundrio/blob/main/model/utils/src/main/java/io/sundr/model/functions/TypeCast.java) investigates the whole inheritance tree to see what type arguments are used on `Map`. 

In your example, the parameters on any reference are ignored and the actual type used is always `Map<Integer, Boolean>`, so that's what is rendered. 
(Except ` WARN Property 'test3' with 'io.fabric8.crd.example.map.ContainingMapsSpec.MyMultiMap<java.lang.String,java.lang.String>' key type is mapped to 'string' because of CRD schemas limitations` :D)

```

  private MyMultiMap<String, String> test3;

  class MyMultiMap<K, V> extends HashMap<Integer, Boolean> {
  }
```

becomes:
```
test3:
  additionalProperties:
    type: boolean
  type: object
```

I will add test covering this and other cases as suggested.